libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/sound/MapboxAudioComponent.kt

```diff
@@ -0,0 +1,61 @@
+package com.mapbox.navigation.dropin.component.sound
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.extensions.flowRoutesUpdated
+import com.mapbox.navigation.dropin.extensions.flowVoiceInstructions
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.updateAndGet
+import kotlinx.coroutines.launch
+
+@ExperimentalPreviewMapboxNavigationAPI
+class MapboxAudioComponent : UIComponent() {
+
+    private val mutableStateFlow = MutableStateFlow(MapboxAudioComponentState())
+
+    fun stateFlow(): StateFlow<MapboxAudioComponentState> = mutableStateFlow
+
+    fun mute() {
+        mutableStateFlow.updateAndGet { state -> state.copy(isMuted = true) }
+    }
+
+    fun unmute() {
+        mutableStateFlow.updateAndGet { state -> state.copy(isMuted = false) }
+    }
+
+    fun toggle() {
+        mutableStateFlow.updateAndGet { state -> state.copy(isMuted = !state.isMuted) }
+    }
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+
+        // TODO handle all of this in an api class.
+        coroutineScope.launch {
+            mapboxNavigation.flowRoutesUpdated().collect {
```

Comment:
It seems weird to me for sound button to observe a route. As explained below why not use `NavigationState` and change the visibility accordingly.